apps/ewallet_db/lib/ewallet_db/transfer.ex

```diff
@@ -172,18 +172,28 @@ defmodule EWalletDB.Transfer do
   using the passed idempotency token.
   """
   def insert(attrs) do
-    changeset = changeset(%Transfer{}, attrs)
     opts = [on_conflict: :nothing, conflict_target: :idempotency_token]
 
-    case Repo.insert(changeset, opts) do
-      {:ok, transfer} ->
-        {:ok, get_by_idempotency_token(transfer.idempotency_token)}
-
-      changeset ->
-        changeset
+    %Transfer{}
+    |> changeset(attrs)
+    |> do_insert(opts)
+    |> case do
+      {_, res} -> res
```

Comment:
Would the new return types cause problem? E.g. before, it would return `{:ok, result}` or `changeset`. Now it's `{:ok, result}` or `{:error, changeset}`?

Reply:
`insert` actually returns `{:error, changeset}`, here it's just shortcuted to `changeset`. [Insert doc](https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert/2)